### PR TITLE
[feature](merge-cloud) Add CloudGlobalTransactionMgr

### PR DESCRIPTION
**Pull request by swjtu-zhanglei**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by SWJTU-ZhangLei** [comment]
run buildall

**Comment by SWJTU-ZhangLei** [comment]
run buildall

**Comment by SWJTU-ZhangLei** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java [hunk 1]
```diff
@@ -688,7 +690,11 @@ public Env(boolean isCheckpointCatalog) {
         this.brokerMgr = new BrokerMgr();
         this.resourceMgr = new ResourceMgr();
 
-        this.globalTransactionMgr = new GlobalTransactionMgr(this);
+        if (Config.isCloudMode()) {
```
Comment (yujun777) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java:693]: EnvFactory.getInstance().createGlobalTransactionMgr(this);

EnvFactory and CloudEnvFactory need implement createGlobalTransactionMgr

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java [hunk 2]
```diff
@@ -2044,6 +2050,10 @@ public long loadAuth(DataInputStream dis, long checksum) throws IOException {
     }
 
     public long loadTransactionState(DataInputStream dis, long checksum) throws IOException {
+        if (Config.isCloudMode()) {
```
Comment (yujun777) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java:2053]: CloudEnv override this function

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java [hunk 3]
```diff
@@ -2358,6 +2368,9 @@ public long saveAuth(CountingDataOutputStream dos, long checksum) throws IOExcep
     }
 
     public long saveTransactionState(CountingDataOutputStream dos, long checksum) throws IOException {
+        if (Config.isCloudMode()) {
```
Comment (yujun777) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java:2371]: CloudEnv overwrite this function

**Review by yujun777 — COMMENTED** [review]
need update

**Comment by SWJTU-ZhangLei** [comment]
run buildall

**Comment by SWJTU-ZhangLei** [comment]
> need update

Thanks for your comments , i have updated the code.

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by gavinchou — APPROVED** [review]

**Review by w41ter — APPROVED** [review]

**Pull request merged**